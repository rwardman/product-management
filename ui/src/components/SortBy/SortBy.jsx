import React, { useState, useEffect } from "react";

import PropTypes from "prop-types";
import * as Styled from "./SortBy.styles";

const SortBy = ({ options, sortBy }) => {
  const [chosenOption, setChosenOption] = useState(options[0]);

  useEffect(() => {
    sortBy(chosenOption);
  }, [sortBy, chosenOption]);

  const handleChange = (e) => {
    const option = e.target.value;
    setChosenOption(option);
  };

  return (
    <Styled.Form>
      <label>
        Sort By:
        <Styled.Select value={chosenOption} onChange={(e) => handleChange(e)}>
          {options.map((option) => (
            <option value={option} key={option}>
              {option}
            </option>
          ))}
        </Styled.Select>
      </label>
    </Styled.Form>
  );
};

export default SortBy;

SortBy.propTypes = {
  options: PropTypes.array.isRequired,
  sortBy: PropTypes.func.isRequired,
};
