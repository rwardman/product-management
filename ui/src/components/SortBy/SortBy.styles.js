import styled from "styled-components";

const Form = styled.form`
  border: 1px solid #aaa;
  border-radius: 3px;
  padding: 5px;
  height: 50px;
  margin-top: 10px;
  @media screen and (min-width: 414px) {
    margin-top: auto;
    margin-bottom: auto;
  }
`;

const Select = styled.select`
  padding: 5px;
  height: 50px;
  margin-left: 5px;
  border-radius: 3px;
  background-color: #fff;
  border: 1px solid #eee;
`;

const Title = styled.span`
  font-size: 16px;
  margin-top: 5px;
  margin-bottom: 5px;
  font-weight: bold;
`;

export { Form, Select, Title };
