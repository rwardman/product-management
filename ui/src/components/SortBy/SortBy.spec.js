import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";

import SortBy from "./SortBy";

// const sortBy = () => {};
const options = ["option1", "option2"];

// test("The sort by component should be displayed correctly", () => {
//   const tree = renderer
//     .create(<SortBy options={options} sortBy={jest.fn()} />)
//     .toJSON();
//   expect(tree).toMatchSnapshot();
// });

describe("When displaying the sort component", () => {
  it("should be displayed correctly", () => {
    const tree = renderer
      .create(<SortBy options={options} sortBy={jest.fn()} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
