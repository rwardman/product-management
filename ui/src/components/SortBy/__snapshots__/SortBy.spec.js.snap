// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When displaying the sort component should be displayed correctly 1`] = `
.c0 {
  border: 1px solid #aaa;
  border-radius: 3px;
  padding: 5px;
  height: 50px;
  margin-top: 10px;
}

.c1 {
  padding: 5px;
  height: 50px;
  margin-left: 5px;
  border-radius: 3px;
  background-color: #fff;
  border: 1px solid #eee;
}

@media screen and (min-width:414px) {
  .c0 {
    margin-top: auto;
    margin-bottom: auto;
  }
}

<form
  className="c0"
>
  <label>
    Sort By:
    <select
      className="c1"
      onChange={[Function]}
      value="option1"
    >
      <option
        value="option1"
      >
        option1
      </option>
      <option
        value="option2"
      >
        option2
      </option>
    </select>
  </label>
</form>
`;
